display.py

```diff
@@ -0,0 +1,53 @@
+from comms import Comms, Message
+from datetime import datetime
+from datetime import time
+import json
+import multiprocessing as mp
+import os
+import random
+import sys
+import threading
+import time
+
+
+class display_base:
+    def __init__(self, fileN):
+        self.comms = Comms()
+        m_list = json.loads(open(fileN).read())
+        self.state = dict()
+        port_number = 4000
+        
+        for item in m_list['mlist']:
+            self.comms.add_subscriber_port(item['ip'],item['port'],item['ip'])
+            self.state[item['ip']] = dict()
+            for key in item['data'].keys():
+                self.state[item['ip']][key] = 0
+        
+        print(self.state)
+
+    def keep_state(self,ip):
+        print("here")
+        while True:
```

Comment:
one thread per robot - This makes sense and will work, but we might want to see what's better in the long run: making one thread that keeps state on everything, or one thread per robot.

Reply:
No need to have an answer now. As long as this works.